lib/src/license_detection/confidence.dart

```diff
@@ -0,0 +1,256 @@
+/// true-negatives are discarded.
+@visibleForTesting
+Range diffRange(String known, List<Diff> diffs) {
+  var foundStart = false;
+  var seen = '';
+  var end = 0;
+  var start = 0;
+
+  for (end = 0; end < diffs.length; end++) {
+    if (seen.length > 1 && seen.trimRight() == known) {
+      break;
+    }
+
+    if (diffs[end].operation == Operation.insert ||
+        diffs[end].operation == Operation.equal) {
+      if (!foundStart) {
+        start = end;
+        foundStart = true;
+      }
+
+      seen += diffs[end].text + ' ';
+    }
+  }
+
+  return Range(start, end);
+}
+
+/// Throws if there was a change with the unknown license.
+void verifyNoVersionChange(Iterable<Diff> diffs, String identifier) {
+  var prevText = '';
+
+  for (var diff in diffs) {
+    final text = diff.text;
+
+    if (diff.operation == Operation.insert) {
+      var number = text;
+      final index = text.indexOf(' ');
+
+      if (index != -1) {
+        number = number.substring(0, index);
+      }
+
+      if (_isVersionNumber(number) && prevText.endsWith('version')) {
+        if (!prevText.endsWith('the standard version') &&
+            !prevText.endsWith('the contributor version')) {
+          throw LicensesMismatchException(-1, identifier);
+        }
+      }
+    } else if (diff.operation == Operation.equal) {
+      prevText = text;
+    }
+  }
+}
+
+///
+void verifyNoGplChange(Iterable<Diff> diffs, String identifier) {
+  var prevText = '';
+  var prevDelete = '';
+
+  for (var diff in diffs) {
+    final text = diff.text;
+
+    switch (diff.operation) {
+      case Operation.insert:
+
+        // Ignores substitution of "library" with "lesser" in gnu license,
+        // but looks for other additions of "lesser" that would lead to
+        // a mismatch.
+        if (text == 'lesser' &&
+            prevText.endsWith('gnu') &&
+            prevDelete != 'library') {
+          if (!prevText.contains('warranty')) {
+            throw LicensesMismatchException(lesserGplChange, identifier);
+          }
+        }
+        break;
+
+      case Operation.delete:
+        if (text == 'lesser' && prevText.endsWith('gnu')) {
+          if (!prevText.contains('warranty')) {
+            throw LicensesMismatchException(lesserGplChange, identifier);
+          }
+        }
+
+        prevDelete = text;
+        break;
+
+      case Operation.equal:
+        prevText = text;
+        prevDelete = '';
+        break;
+    }
+  }
+}
+
+/// Returns a score indicating to what extent the [diffs] are acceptable.
+///
+/// The integer returned indicates the Levenshtein edit distance
+/// calculated on the number of words altered instead of
+/// the number of characters altered.
+@visibleForTesting
+int scoreDiffs(Iterable<Diff> diffs) {
+  return diffLevenshteinWord(diffs);
+}
+
+final _notVersionNumber = RegExp(r'[^0-9\.]');
+
+bool _isVersionNumber(String text) {
+  return !_notVersionNumber.hasMatch(text);
```

Comment:
Maybe explain a bit, like:
```suggestion
  // Checks if [text] contains anything other than 0-9 and `.`
  return !_notVersionNumber.hasMatch(text);
```